### DISCUSSION TITLE
method `doit` is private

### DISCUSSION THREAD

**Discussion by anil-e** [Byron/google-apis-rs#457] [Q&A] [answered]
Hi,

First of all, thank you for making this comprehensive library available.

Unfortunately, I encountered a problem during my test of the Gmail import.

I am trying to use the `messages_import` function as described in the docs: 

```
use gmail1::{ hyper, hyper_rustls, oauth2, Gmail};
use gmail1::{Error, Result as GResult}; 
  
... 

let mut hub = Gmail::new(...);
let result = hub.users().messages_import(req, "anil@termitel.de").doit().await;

```

Unfortunately, I encounter the error: 
```
method `doit` is private
```

Is this a bug or am I using the function incorrectly?

**Comment by Byron** [comment, marked as answer]
> First of all, thank you for making this comprehensive library available.

Thanks, but I find it untenable that Google doesn't provide their own, and the support that would be needed for it.

You probably need to use the [upload()](https://docs.rs/google-gmail1/latest/google_gmail1/api/struct.UserMessageImportCall.html#method.upload) call.

**Comment by anil-e** [comment]
Thanks for the answer @Byron.
I think I got confused, because in the readme first `messages_import()`, with `doit()` is used and then further down correctly without `doit()`.

Since `UserMessageImportCall` does not seem to have a `doit()` method, I think it would make sense to adjust the example in the readme. I have created a pull request for it [here](https://github.com/Byron/google-apis-rs/pull/458) :)
